bundle.Dockerfile

```diff
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=opentelemetry-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.16.0+git
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.17.0
```

Comment:
@erichsueh3 please install operator-sdk 1.17.0 and re-generate the manifest files. This will fix the issue with CI..

Reply:
Hi @pavolloffay, these manifest files were generated after installing operator-sdk 1.17.0; I've done it again just to make sure, and the generated manifest files remain the same

Reply:
How did you run the generation? 

The following command does not change the labels in `bundle.Dockerfile` and `bundle/metadata/annotations.yaml`.
```
make bundle generate api-docs USER=open-telemetry VERSION=0.42.0
```

Reply:
I ran `make bundle USER=open-telemetry VERSION=0.44.0`; this is what step 2 of [RELEASE.md](https://github.com/open-telemetry/opentelemetry-operator/blob/main/RELEASE.md) instructs

Reply:
Thanks for the confirmation.

Please revert this change the label should be `LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.16.0+git` if you run the bundle it will get changed to `v1.16.0+git`.

Reply:
Fixed